app/Http/Controllers/DepartureReleaseController.php

```diff
@@ -0,0 +1,191 @@
+        Request $request,
+        DepartureReleaseRequest $departureReleaseRequest
+    ): JsonResponse {
+        $validated = $request->validate(
+            [
+                'controller_position_id' => 'required|integer',
+            ]
+        );
+
+        $responseData = null;
+        try {
+            $this->departureReleaseService->rejectReleaseRequest(
+                $departureReleaseRequest,
+                $validated['controller_position_id'],
+                Auth::id()
+            );
+            $responseCode = 200;
+        } catch (DepartureReleaseDecisionNotAllowedException $decisionNotAllowedException) {
+            Log::warning(
+                sprintf(
+                    'User %d attempted to reject release %d without permission',
+                    Auth::id(),
+                    $departureReleaseRequest->id
+                )
+            );
+            $responseCode = 422;
+            $responseData = ['message' => 'You cannot reject this release'];
+        }
+        return response()->json($responseData, $responseCode);
+    }
+
+    public function acknowledgeReleaseRequest(
+        Request $request,
+        DepartureReleaseRequest $departureReleaseRequest
+    ): JsonResponse {
+        $validated = $request->validate(
+            [
+                'controller_position_id' => 'required|integer',
+            ]
+        );
+
+        $responseData = null;
+        try {
+            $this->departureReleaseService->acknowledgeReleaseRequest(
+                $departureReleaseRequest,
+                $validated['controller_position_id'],
+                Auth::id()
+            );
+            $responseCode = 200;
+        } catch (DepartureReleaseDecisionNotAllowedException $decisionNotAllowedException) {
+            Log::warning(
+                sprintf(
+                    'User %d attempted to reject release %d without permission',
+                    Auth::id(),
+                    $departureReleaseRequest->id
+                )
+            );
+            $responseCode = 422;
+            $responseData = ['message' => 'You cannot reject this release'];
+        }
+        return response()->json($responseData, $responseCode);
+    }
+
+    public function cancelReleaseRequest(
+        DepartureReleaseRequest $departureReleaseRequest
+    ): JsonResponse {
+        $responseData = null;
+        try {
+            $this->departureReleaseService->cancelReleaseRequest(
+                $departureReleaseRequest,
+                Auth::id()
+            );
+            $responseCode = 200;
+        } catch (DepartureReleaseDecisionNotAllowedException $decisionNotAllowedException) {
+            Log::warning(
+                sprintf(
+                    'User %d attempted to cancel release %d without permission',
+                    Auth::id(),
+                    $departureReleaseRequest->id
+                )
+            );
+            $responseCode = 422;
```

Comment:
Hmmmmm just thinking - but to be honest this only affects you anyway - the 422 status code will have multiple paths via the validator and further via this error here. Is it possibly best served to use something like a 403?

Reply:
Returning 403 now :)